javav2/example_code/redshift/src/main/java/com/example/scenario/RedshiftScenario.java

```diff
@@ -0,0 +1,550 @@
+                .build();
+
+            ModifyClusterResponse clusterResponse = redshiftClient.modifyCluster(modifyClusterRequest);
+            System.out.println("The modified cluster was successfully modified and has "
+                + clusterResponse.cluster().preferredMaintenanceWindow() + " as the maintenance window");
+
+        } catch (RedshiftException e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+    }
+    // snippet-end:[redshift.java2.mod_cluster.main]
+
+    // snippet-start:[redshiftdata.java2.query.main]
+    public static String queryMoviesByYear(RedshiftDataClient redshiftDataClient,
+                                           String database,
+                                           String dbUser,
+                                           String sqlStatement,
+                                           String clusterId) {
+
+        try {
+            ExecuteStatementRequest statementRequest = ExecuteStatementRequest.builder()
+                .clusterIdentifier(clusterId)
+                .database(database)
+                .dbUser(dbUser)
+                .sql(sqlStatement)
+                .build();
+
+            ExecuteStatementResponse response = redshiftDataClient.executeStatement(statementRequest);
+            return response.id();
+
+        } catch (RedshiftDataException e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+        return "";
+    }
+    // snippet-end:[redshiftdata.java2.query.main]
+
+    // snippet-start:[redshiftdata.java2.getresults.main]
+    public static void getResults(RedshiftDataClient redshiftDataClient, String statementId) {
+        try {
+            GetStatementResultRequest resultRequest = GetStatementResultRequest.builder()
+                .id(statementId)
+                .build();
+
+            // Extract and print the field values using streams.
+            GetStatementResultResponse response = redshiftDataClient.getStatementResult(resultRequest);
+            response.records().stream()
+                .flatMap(List::stream)
+                .map(Field::stringValue)
+                .filter(value -> value != null)
+                .forEach(value -> System.out.println("The Movie title field is " + value));
+
+        } catch (RedshiftDataException e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+    }
+    // snippet-end:[redshiftdata.java2.getresults.main]
+
+    // snippet-start:[redshift.java2.describe_cluster.main]
+    public static void waitForClusterReady(RedshiftClient redshiftClient, String clusterId) {
+        boolean clusterReady = false;
+        String clusterReadyStr;
+        System.out.println("Waiting for cluster to become available. This may take a few mins.");
+        try {
+            DescribeClustersRequest clustersRequest = DescribeClustersRequest.builder()
```

Comment:
Shouldn't describeClusters use a paginator?

Reply:
In this case, where 1 cluster is being returned, not required.